src/api/mixPanel.ts

```diff
@@ -0,0 +1,92 @@
+import { Agent } from "..";
+
+export interface AnalyticsOptions {
+    mixPanelKey: string;
+    agent: Agent;
+    chatId?: string;
+    isEnabled?: boolean;
+    distinct_id?: string;
+}
+
+class AnalyticsProvider {
+    private static instance: AnalyticsProvider;
+    private mixPanelKey: string;
+    private distinct_id?: string;
+    private isEnabled: boolean = true;
+    private chatId?: string;
+    private agentId: string;
+    private owner_id: string;
+
+    private constructor(config: AnalyticsOptions) {
+        this.mixPanelKey = config.mixPanelKey || 'testKey';
+        this.distinct_id = config.distinct_id || this.getUUID();
```

Comment:
we said its supposed to be external id, is it being injected from outside or anything?

Reply:
distinctId is now optional param if SDK create params AgentManagerOptions
So, if it's passed to sdk, will be taken from there

Reply:
should always be external_id